### DISCUSSION TITLE
[Feature Request] - Deletion of content between environments to happen with "transfers" or detectable between environments

### DISCUSSION THREAD

**Discussion by closdesign** [umbraco/Umbraco.Cloud.Issues#893] [Ideas]
### Issue description

I guess we put feature requests here?

It is slighty a bummer, actually, more annoying, when we have deletion of content between environments they aren't picked up by a "Content Transfer" of a parent node or a "Partial Restore" of a parent node to environments.
Can only be done manually between environments.  Which is slightly frustrating, because we have our 20+ content authors locked off from our production environment so they can't delete the site.  
https://docs.umbraco.com/umbraco-cloud/deployments/deploying-deletions

I would be super helpful to have the ability to detect deletions between environments for content.

Thank you

**Comment by ronaldbarendse** [comment]
Hi @closdesign, thanks for the suggestion and apologies for the late reply.

This isn't specific to Umbraco Cloud: deployments between environments are handled by Umbraco Deploy, which is also used on-premises. Feature requests for it belong in the [Umbraco Deploy issue tracker](https://github.com/umbraco/Umbraco.Deploy.Issues), so I'm closing this discussion. If you still want this considered after reading the below, please open a new discussion there.

Not deploying deletions of content is by design. A content transfer or partial restore only adds or updates the items you select on the target environment. Removing content that exists on the target but not on the source can't be distinguished from content that was created on the target and never transferred back (for example, editors working directly on Live). Silently deleting it would be an unrecoverable data loss, which is why it requires an explicit action on each environment. The [Deploying Deletions](https://docs.umbraco.com/umbraco-cloud/build-and-customize-your-solution/handle-deployments-and-environments/deployment/deploying-deletions) article explains this in more detail.

For schema (Document Types, Data Types, Templates, Languages, etc.) this has improved since you opened this. Umbraco Deploy 13.4 and later can clean the schema: items in the database that don't have a corresponding UDA file on disk are removed. You can run this manually via the **Verify schema** operation on the [Deploy Dashboard](https://docs.umbraco.com/umbraco-cloud/build-and-customize-your-solution/handle-deployments-and-environments/deployment/deploy-dashboard), or have it run automatically after every schema deployment by setting [`PostDeploySchemaOperation`](https://docs.umbraco.com/umbraco-deploy/getting-started/deploy-settings#post-deploy-schema-operation) to `CleanSchema`. Be careful with the automatic option on Live, as deleting a Document Type also deletes all content using it. The Deploying Deletions article will be updated to cover this, as it's currently missing there.

For your content workflow, a couple of alternatives to consider:

- If the main reason for keeping editors off Live is to prevent accidental deletions, you can restrict the Delete permission using [granular Document permissions](https://docs.umbraco.com/umbraco-cms/manage-and-publish-content/users-and-members/users#granular-permissions) on specific nodes, such as the root nodes. Editors can then work directly on Live, which avoids having to keep content in sync across environments.
- If you need review and approval before content goes live, [Umbraco Workflow](https://docs.umbraco.com/umbraco-workflow) provides that on a single environment. Using multiple environments for content review is usually more work than an approval workflow, since environments are primarily intended for schema and code changes.
